packages/@n8n/task-runner-python/src/env.py

```diff
@@ -0,0 +1,73 @@
+import os
+from typing import Set
+
+from src.constants import (
+    DEFAULT_MAX_CONCURRENCY,
+    DEFAULT_TASK_TIMEOUT,
+    DEFAULT_TASK_BROKER_URI,
+    DEFAULT_MAX_PAYLOAD_SIZE,
+    DEFAULT_DENIED_BUILTINS,
+    ENV_MAX_CONCURRENCY,
+    ENV_MAX_PAYLOAD_SIZE,
+    ENV_TASK_BROKER_URI,
+    ENV_GRANT_TOKEN,
+    ENV_TASK_TIMEOUT,
+    ENV_BUILTINS_DENY,
+    ENV_STDLIB_ALLOW,
+    ENV_EXTERNAL_ALLOW,
+)
+from src.task_runner import TaskRunnerOpts
+
+
+def parse_allowlist(allowlist_str: str, list_name: str) -> Set[str]:
+    if not allowlist_str:
+        return set()
+
+    modules = {
+        module
+        for raw_module in allowlist_str.split(",")
+        if (module := raw_module.strip())
+    }
+
+    if "*" in modules and len(modules) > 1:
+        raise ValueError(
+            f"Wildcard '*' in {list_name} must be used alone, not with other modules. "
+            f"Got: {', '.join(sorted(modules))}"
+        )
+
+    return modules
+
+
+def parse_env_vars() -> TaskRunnerOpts:
+    grant_token = os.getenv(ENV_GRANT_TOKEN, "")
+
+    if not grant_token:
+        raise ValueError(f"{ENV_GRANT_TOKEN} environment variable is required")
+
+    denied_builtins_str = os.getenv(ENV_BUILTINS_DENY, DEFAULT_DENIED_BUILTINS)
+    denied_builtins = {
+        name
+        for raw_name in denied_builtins_str.split(",")
+        if (name := raw_name.strip())
+    }
```

Comment:
nit:

Could also extract this to a function to make it match the allow handling and make it cleaner to read

```suggestion
    denied_builtins = parse_denied_list(denied_builtins_str)
```

Reply:
[1879d44](https://github.com/n8n-io/n8n/pull/18826/commits/1879d44c5839675c0f14b9faf18ac49bfe0d7d14)